File: forge/src/main/java/dev/gluton/quickspyglasser/forge/QuickSpyglasserForge.java
package dev.gluton.quickspyglasser.forge;

import dev.architectury.platform.forge.EventBuses;
import dev.gluton.quickspyglasser.QuickSpyglasser;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(QuickSpyglasser.MOD_ID)
public class QuickSpyglasserForge {
    public QuickSpyglasserForge() {
        // Submit our event bus to let architectury register our content on the right time
        EventBuses.registerModEventBus(QuickSpyglasser.MOD_ID, FMLJavaModLoadingContext.get().getModEventBus());
        QuickSpyglasser.init();
    }
}

File: forge/src/main/kotlin/dev/gluton/quickspyglasser/forge/QuickSpyglasserForge.kt
package dev.gluton.quickspyglasser.forge

import dev.architectury.platform.forge.EventBuses
import dev.gluton.quickspyglasser.QuickSpyglasser
import dev.gluton.quickspyglasser.client.QuickSpyglasserClient
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import thedarkcolour.kotlinforforge.forge.runForDist

@Mod(QuickSpyglasser.ID)
object QuickSpyglasserForge {
	init {
		EventBuses.registerModEventBus(QuickSpyglasser.ID, MOD_BUS)

		runForDist(
			clientTarget = {
				MOD_BUS.addListener { _: FMLClientSetupEvent -> QuickSpyglasserClient.init() }
			},
			serverTarget = {}
		)
	}
}
